src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHandlerManager.cs

```diff
@@ -73,7 +90,11 @@ private void HandleEvaluation(IProjectVersionedValue<IProjectSubscriptionUpdate>
                     IProjectChangeDescription projectChange = update.Value.ProjectChanges[ruleName];
                     if (projectChange.Difference.AnyChanges)
                     {
-                        handler.handler.Handle(version, projectChange, isActiveContext, logger);
+                        _currentHandlerTask = _currentHandlerTask.ContinueWith(
```

Comment:
Is there a race here? What if both threads are in ContinueWith at same time?

Reply:
I think the only race is in `HandleDesignTime`, that another project could queue something in between the 3 tasks that happen there.  Otherwise, isn't reference assignment atomic on x86?

Reply:
There is no race because these were already protected by the lock in `LanguageServiceHost.HandleAsync`.